gcloud/datastore/entity.py

```diff
@@ -77,6 +77,32 @@ def __init__(self, key=None, exclude_from_indexes=()):
         self.key = key
         self._exclude_from_indexes = set(exclude_from_indexes)
 
+    def __eq__(self, other):
+        """Compare two entities for equality.
+
+        Entities compare equal if their keys compare equal, and their
+        properties compare equal.
+
+        :rtype: boolean
+        :returns: True if the entities compare equal, else False.
+        """
+        if not isinstance(other, Entity):
+            raise NotImplementedError
```

Comment:
Per the [Python data model docs](https://docs.python.org/2/reference/datamodel.html#the-standard-type-hierarchy), I think it should actually be:

```python
    return NotImplemented
```

which would allow the interpreter to test `other` (in case it had the know-how to make the comparison).